frontend/src/components/editor/navigation/multi-cell-action-toolbar.tsx

```diff
@@ -0,0 +1,322 @@
+      },
+      {
+        icon: <XCircleIcon size={13} strokeWidth={1.5} />,
+        label: "Clear outputs",
+        handle: clearSelectedCellsOutput,
+      },
+    ],
+    [
+      {
+        icon: <EyeOffIcon size={13} strokeWidth={1.5} />,
+        label: "Hide cells",
+        handle: (cellIds) =>
+          toggleSelectedCellsProperty(cellIds, "hide_code", true),
+      },
+      {
+        icon: <EyeIcon size={13} strokeWidth={1.5} />,
+        label: "Show cells",
+        handle: (cellIds) =>
+          toggleSelectedCellsProperty(cellIds, "hide_code", false),
+      },
+    ],
+    [
+      {
+        icon: <ZapOffIcon size={13} strokeWidth={1.5} />,
+        label: "Disable cells",
+        handle: (cellIds) =>
+          toggleSelectedCellsProperty(cellIds, "disabled", true),
+      },
+      {
+        icon: <ZapIcon size={13} strokeWidth={1.5} />,
+        label: "Enable cells",
+        handle: (cellIds) =>
+          toggleSelectedCellsProperty(cellIds, "disabled", false),
+      },
+    ],
+  ];
+
+  // Filter out hidden actions and empty groups
+  return {
+    actions: actions
+      .map((group) => group.filter((action) => !action.hidden))
+      .filter((group) => group.length > 0),
+    moreActions: moreActions
+      .map((group) => group.filter((action) => !action.hidden))
+      .filter((group) => group.length > 0),
+  };
+}
+
+export const MultiCellActionToolbar = () => {
+  const selectionState = useCellSelectionState();
+  const selectedCells = [...selectionState.selected];
+
+  if (selectedCells.length < 2) {
+    return null;
+  }
+
+  return <MultiCellActionToolbarInternal cellIds={selectedCells} />;
+};
+
+const Separator = () => <div className="h-4 w-px bg-border mx-1" />;
+
+const MultiCellActionToolbarInternal = ({ cellIds }: { cellIds: CellId[] }) => {
+  const selectionActions = useCellSelectionActions();
+  const { actions, moreActions } = useMultiCellActionButtons(cellIds);
+
+  const selectedCount = cellIds.length;
+
+  useEventListener(window, "mousedown", (evt) => {
+    // Clear selected, unless clicked inside an element that contains data-keep-cell-selection
+    if (
+      (evt.target instanceof HTMLElement || evt.target instanceof SVGElement) &&
+      evt.target.closest("[data-keep-cell-selection]") !== null
+    ) {
+      return;
+    }
+
+    // HACK: evt.target is the root <html> element
+    // when the DropdownMenu is closed.
+    // This prevents the selection from being cleared when the opens/closes
+    // the DropdownMenu.
+    if (evt.target instanceof HTMLHtmlElement) {
+      return;
+    }
+    selectionActions.clear();
+  });
+
+  return (
+    <div
+      className="fixed bottom-12 justify-center flex w-full left-0 right-0 z-50"
+      data-keep-cell-selection={true}
+    >
+      <div className="bg-background/95 backdrop-blur supports-[backdrop-filter]:bg-background/60 border border-input rounded-lg shadow-lg p-2 overflow-x-auto overflow-y-hidden mx-20 scrollbar-thin">
```

Comment:
Let's use slate-7 for the border, we use it elsewhere throughout the app too.

```suggestion
      <div className="bg-background/95 backdrop-blur supports-[backdrop-filter]:bg-background/60 border border-[var(--slate-7)] rounded-lg shadow-lg p-2 overflow-x-auto overflow-y-hidden mx-20 scrollbar-thin">
```